### PR TITLE
Fix bug 1536670 - Update six to 1.12.0

### PR DESCRIPTION
**Pull request by jotes**
Hey,

I updated `six` to 1.12.0 and replaced all missing methods with their equivalents.

@adngdb do you have a moment to look at this?

### CODE REVIEW

File: pontoon/checks/libraries/pontoon.py [hunk 1]
```diff
@@ -33,7 +35,7 @@ def run_checks(entity, string):
     Group all checks related to the base UI
 
     :arg pontoon.base.models.Entity entity: Source entity
-    :arg basestring string: a translation
+    :arg six.string_types string: a translation
```
Comment (jotes) [inline pontoon/checks/libraries/pontoon.py]: @adngdb Do you prefer the previous version?

Comment (adngdb) [reply pontoon/checks/libraries/pontoon.py]: I think it would better to be less specific here. Hopefully once we're moved to py3 we should remove `six`, right?

File: pontoon/localizations/views.py [hunk 2]
```diff
@@ -89,7 +89,7 @@ def ajax_resources(request, code, slug):
         pages[page.name] = latest_page_translatedresource
 
     translatedresources = {s.resource.path: s for s in translatedresources_qs}
-    translatedresources = dict(translatedresources.items() + pages.items())
+    translatedresources = dict(list(translatedresources.items()) + pages.items())
```
Comment (adngdb) [inline pontoon/localizations/views.py]: I've seen a few of those `list()` added. Why is that needed?

Comment (Pike) [reply pontoon/localizations/views.py]: You can't concat `dict_items`. Interestingly enough, this is not enough, both `items` need to be converted.

Or you create a dict from `translatedresources` first, and then update from `pages`.

Comment (adngdb) [reply pontoon/localizations/views.py]: Ah, `.keys()` doesn't return a simple `list`? I didn't know that. Thanks!

Comment (jotes) [reply pontoon/localizations/views.py]: @Pike Thanks :) Added to my todo list.

File: pontoon/checks/libraries/pontoon.py (hunk 1 above)
Comment (jotes) [reply pontoon/checks/libraries/pontoon.py]: That's right, I'll revert it then.
